### ISSUE TITLE
The inner library used by mermaid seems to support node ranking (https://github.com/dagrejs/dagre/wiki#configuring-the-layout) through the `graph.ranker` value. Currently the only supported value by Mermaid is `rankdir` (TB, LR, RL...), but adding different props to the `graph` key could work.

### ISSUE THREAD

**Issue by polymathuk** [mermaid-js/mermaid#1736]
The inner library used by mermaid seems to support node ranking (https://github.com/dagrejs/dagre/wiki#configuring-the-layout) through the `graph.ranker` value. Currently the only supported value by Mermaid is `rankdir` (TB, LR, RL...), but adding different props to the `graph` key could work.

_Originally posted by @hugmanrique in https://github.com/mermaid-js/mermaid/issues/270#issuecomment-439637473_

**Comment by polymathuk** [comment]
Yet another layout options request. I see from #270 that there are many alignment and other useful formatting arguments that can be applied in the underlying renderer. Can we have a feature in Mermaid that it passes these arguments from the markup to the renderer?
Presently we can use:

```
graph TB
    A[node A]-->B[node B]
```

What would be handy would be something like:

```
graph rankdir:TB, align:UL,
    A[node A] edge weight:1 --> B[node B]
```

Is this possible?

**Comment by GDFaber** [comment]
Out of curiosity I did some testing on the align parameter of dagre and it worked pretty well in some use cases, maybe not all. But integrating this parameter should be possible. Don't know about the weight options though.

**Comment by knsv** [comment]
Good idea. I have been thinking about this as well but never really got around to it. When this is implemented it should be done mainly for flowchart as that is the future of flowcharts, not graph.
